src/zspec/zcl/definition/tstype.ts

```diff
@@ -95,6 +95,7 @@ export interface Cluster {
 
 export interface ClusterDefinition {
     ID: number;
+    name?: string;
```

Comment:
```suggestion
    name: string;
```

I propose to make this mandatory

Reply:
Making that mandatory would enforce duplicating the name in code for all ZCL clusters (uses the object key to add the name at runtime at the moment). That's why `Cluster` has `name` but not `ClusterDefinition`.

Reply:
The reason i didn't put it mandatory is because we will need to update the CustomCluster that are currently rolling out. In In the current implementation, the actual name of the cluster is used as a key of the object, rather than an attribute of it. 

Which i guess is used to enforced some form of primary key.

Example of current definition model:
```
const myClustersDefinition: {
    [s: string]: ClusterDefinition
} = {
    **manuSpecificCluster**: {
        ID: 0xFC01,
        manufacturerCode: Zcl.ManufacturerCode.MANUF,
        attributes: {},
        commands: {},
        commandsResponse: {},
},
etc ... 
```

Update : @Nerivec  answered at the same time :)

Reply:
@Nerivec @Koenkk The reason of this change is because i was looking for an data-only representation of a Cluster object to use in the frontend UI.

- `Cluster` interface enforce implementing some methods
- `ClusterDefinition` interface is lacking the `name` attribute

Name optional was an ok-inbetween.

Reply:
That also means frontend would remain dependent on the herdsman package since it would need the types import?

Reply:
Yes.. now that @Koenkk has clarified what we do with https://github.com/Koenkk/zigbee2mqtt/pull/22583, i will make the necessary changes in order to remove the dependency completely.